internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -570,13 +570,16 @@ func (c *Coordinator) PerformComponentDiagnostics(ctx context.Context, additiona
 
 // SetLogLevel changes the entire log level for the running Elastic Agent.
 // Called from external goroutines.
-func (c *Coordinator) SetLogLevel(ctx context.Context, lvl logp.Level) error {
+func (c *Coordinator) SetLogLevel(ctx context.Context, lvl *logp.Level) error {
```

Comment:
Why change this to allow a nil value, to then error if a nil value is passed?

You also convert the pointer to value in the function. Seems to me all the changes in this function could be removed and it would have the same result.

Reply:
Mostly it is to have a single interface for log level setter and then leaving the check to the implementations...
If I didn't change this signature I would have to define 2 interfaces: one with a pointer (for setting the log level coming from policy which could be cleared) and one with the value implemented by coordinator (which needs to receive a real value)

The signal to noise ration of having 2 interfaces with similar names differing only by a pointer seemed a bit much to me but I still wanted to abstract away the coordinator package so this is the tradeoff I came to.

There's no technical reason why we could not have 2 separate interfaces though...

Reply:
No that makes complete sense. I was just confused on the change, but if that unifies it I prefer that.